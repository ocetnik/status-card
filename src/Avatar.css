.Image-wrapper {
  padding: 10px 12px;
}

.Image {
  background-position: center center;
  background-size: cover;
  border-radius: 50%;
  position: relative;
  height: 32px;
  width: 32px;
}
