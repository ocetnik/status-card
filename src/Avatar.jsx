import React from 'react';

import './Avatar.css';

export default ({ imageUrl, children }) =>
  imageUrl ? (
    <div className="Image-wrapper">
      <div className="Image" style={{ backgroundImage: `url(${imageUrl})` }}>
        {children}
      </div>
    </div>
  ) : (
    children
  );
