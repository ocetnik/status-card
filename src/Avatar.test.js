import React from 'react';
import renderer from 'react-test-renderer';

import Avatar from './Avatar';

it('renders with image', () => {
  const tree = renderer
    .create(<Avatar children={<div>badge</div>} imageUrl={'image-url'} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders without image', () => {
  const tree = renderer.create(<Avatar children={<div>badge</div>} />).toJSON();
  expect(tree).toMatchSnapshot();
});
