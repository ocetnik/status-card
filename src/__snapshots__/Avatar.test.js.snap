// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with image 1`] = `
<div
  className="Image-wrapper"
>
  <div
    className="Image"
    style={
      Object {
        "backgroundImage": "url(image-url)",
      }
    }
  >
    <div>
      badge
    </div>
  </div>
</div>
`;

exports[`renders without image 1`] = `
<div>
  badge
</div>
`;
